src/bin/pg_autoctl/pgctl.c

```diff
@@ -2046,6 +1917,69 @@ prepare_conninfo_sslmode(PQExpBuffer buffer, SSLOptions sslOptions)
 }
 
 
+/*
+ * pgctl_identify_system connects with replication=1 to our target node and run
+ * the IDENTIFY_SYSTEM command to check that HBA is ready.
+ */
+bool
+pgctl_identify_system(ReplicationSource *replicationSource)
+{
+	NodeAddress *primaryNode = &(replicationSource->primaryNode);
+
+	char primaryConnInfo[MAXCONNINFO] = { 0 };
+	char primaryConnInfoReplication[MAXCONNINFO] = { 0 };
+	PGSQL replicationClient = { 0 };
+	int len = 0;
+
+	/* sometimes we don't have a primary, just pretend everything is fine */
+	if (IS_EMPTY_STRING_BUFFER(primaryNode->host))
+	{
+		return true;
+	}
+
+	if (!prepare_primary_conninfo(primaryConnInfo,
+								  MAXCONNINFO,
+								  primaryNode->host,
+								  primaryNode->port,
+								  replicationSource->userName,
+								  NULL, /* no database */
+								  replicationSource->password,
+								  replicationSource->applicationName,
+								  replicationSource->sslOptions,
+								  false)) /* no need for escaping */
+	{
+		/* errors have already been logged. */
+		return false;
+	}
+
+	len = sformat(primaryConnInfoReplication, MAXCONNINFO,
```

Comment:
Minor: I guess this information is not likely that  every reader of the code is aware of. Maybe add a comment from the docs: `A Boolean value of true (or on, yes, 1) tells the backend to go into physical replication walsender mode.`

Reply:
I added an extract from the Postgres docs to explain what we're doing here.